tree/dataframe/inc/ROOT/RDF/InterfaceUtils.hxx

```diff
@@ -501,6 +509,55 @@ void JitDefineHelper(F &&f, const char **colsPtr, std::size_t colsSize, std::str
    delete wkJittedDefine;
 }
 
+template <typename F>
+void JitVariationHelper(F &&f, const char **colsPtr, std::size_t colsSize, const char **variedCols,
+                        std::size_t variedColsSize, const char **variationTags, std::size_t variationTagsSize,
+                        std::string_view variationName, RLoopManager *lm,
+                        std::weak_ptr<RJittedVariation> *wkJittedVariation, RColumnRegister *colRegister,
+                        std::shared_ptr<RNodeBase> *prevNodeOnHeap) noexcept
+{
+   if (wkJittedVariation->expired()) {
+      // The branch of the computation graph that needed this jitted variation went out of scope between the type
+      // jitting was booked and the time jitting actually happened. Nothing to do other than cleaning up.
+      delete wkJittedVariation;
+      // colRegister must be deleted before prevNodeOnHeap because their dtor needs the RLoopManager to be alive
+      // and prevNodeOnHeap is what keeps it alive if the rest of the computation graph is already out of scope
```

Comment:
lifetime management is awkward because to make RDF's deferred jitting work we are putting stuff on the heap that normally wouldn't be, and as a consequence it could outlive everything else in the computation graph, which normally it could not. I'll see whether we can do better.